assets/differentialabundance_report.Rmd

```diff
@@ -733,16 +733,21 @@ for (i in 1:nrow(contrasts)){
     cat("\n##### ", pvt, " p values\n")
     pval_column <- p_value_types[[pvt]]
   
-    full_de$differential_status <- FALSE
-    full_de$differential_status[abs(full_de[[params$differential_fc_column]]) > log2(params$differential_min_fold_change) & full_de[[pval_column]] < p_value_thresholds[[pvt]]] <- TRUE
+    full_de$differential_status <- "Not significant"
+    full_de$differential_status[abs(full_de[[params$differential_fc_column]]) > log2(params$differential_min_fold_change)] <- paste("abs(logFC) >", params$differential_min_fold_change)
+    full_de$differential_status[full_de[[pval_column]] < p_value_thresholds[[pvt]]] <- paste(pvt, "<=", p_value_thresholds[[pvt]])
+    full_de$differential_status[abs(full_de[[params$differential_fc_column]]) > log2(params$differential_min_fold_change) & full_de[[pval_column]] < p_value_thresholds[[pvt]]] <- paste("abs(logFC) >", params$differential_min_fold_change, "&", pvt, "<=", p_value_thresholds[[pvt]])
 
     # Define the thresholds we'll draw
 
     hline_thresholds = vline_thresholds = list()
     hline_thresholds[[paste(pval_column, '=', p_value_thresholds[[pvt]])]] = -log10(p_value_thresholds[[pvt]])
     vline_thresholds[[paste(params$differential_fc_column, '<-', log2(params$differential_min_fold_change))]] = -log2(params$differential_min_fold_change)
     vline_thresholds[[paste(params$differential_fc_column, '>', log2(params$differential_min_fold_change))]] = log2(params$differential_min_fold_change)
-
+    
+    palette_volcano <- makeColorScale(4, params$differential_palette_name)
+    names(palette_volcano) <- c("Not significant", paste("abs(logFC) >", params$differential_min_fold_change), paste(pvt, "<=", p_value_thresholds[[pvt]]), paste("abs(logFC) >", params$differential_min_fold_change, "&", pvt, "<=", p_value_thresholds[[pvt]]))
```

Comment:
```suggestion
```
I don't think naming the palette makes a difference- the function just receives a vector of colors (the names come from the factor levels)